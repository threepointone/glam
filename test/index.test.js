/* eslint-disable jsx-quotes,no-useless-escape */
/* eslint-env jest */

const fs = require('fs')
const path = require('path')

const css = require('../src').default

const { sheet, flush } = require('../src')

test('returns a class for a string', () => {
  expect(css`color:red`).toMatchSnapshot()
})

test('generates readable classnames', () => {
  expect(css` color: red; name: whatup;`).toMatchSnapshot()
})

test('returns 2 classes for a dynamic string', () => {
  expect(css`color:red, font-weight:${'bold'}`).toMatchSnapshot()
})

test('receives a class and array of var values', () => {
  let css = function (a, b) {
    return [a, b]
  }
  expect(css`color:red, font-weight:${'bold'}`).toMatchSnapshot()
})

test('requires css file corresponding to module', () => {
  expect(global.mockedCssImports['index.test.js.css']).toMatchSnapshot()
})

test('injects dynamic values into a sheet', () => {
  flush()
  let cls = css`color:red; font-weight:${'bold'};flex:1 0 0;`
  expect(sheet.rules()).toMatchSnapshot()
})

test('dudupe static sections', () => {
  flush()
  let cls1 = `color:red, font-weight:${'bold'}`
  let cls2 = `color:red, font-weight:${'normal'}`
  expect(cls1.split(' ')[0]).toBe(cls2.split(' ')[0])
  expect([cls1, cls2].map(x => x.split(' ')[1])).toMatchSnapshot()
})

test(`extracts css into a css file`, () => {
  let file = path.join(__dirname, './index.test.js.css')
  expect(fs.existsSync(file)).toBe(true)
  expect(fs.readFileSync(file, 'utf8')).toMatchSnapshot()
})
