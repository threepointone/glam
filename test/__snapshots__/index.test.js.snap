// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dudupe static sections 1`] = `
Array [
  "font-weight:bold",
  "font-weight:normal",
]
`;

exports[`extracts css into a css file 1`] = `
"/* do not edit this file */
.css-hwfcu5 { color: red }
.css-whatup-1xwk7zl { color: red; name: whatup; }
.css-1nd0lsp { color: red,; font-weight: var(--css-1nd0lsp-0) }
.css-gjw345 { color: red; font-weight: var(--css-gjw345-0);-webkit-flex: 1 0 0;-ms-flex: 1 0 0;flex: 1 0 0; }"
`;

exports[`generates readable classnames 1`] = `"css-whatup-1xwk7zl"`;

exports[`injects dynamic values into a sheet 1`] = `Array []`;

exports[`receives a class and array of var values 1`] = `
Array [
  "css-1nd0lsp",
  Array [
    "bold",
  ],
]
`;

exports[`requires css file corresponding to module 1`] = `
"/* do not edit this file */
.css-hwfcu5 { color: red }
.css-1nd0lsp { color: red,; font-weight: var(--css-1nd0lsp-0) }
.css-gjw345 { color: red; font-weight: var(--css-gjw345-0);-webkit-flex: 1 0 0;-ms-flex: 1 0 0;flex: 1 0 0; }"
`;

exports[`returns 2 classes for a dynamic string 1`] = `"css-1nd0lsp vars-1wb9cdj"`;

exports[`returns a class for a string 1`] = `"css-hwfcu5"`;
